llms/mlx_lm/generate.py

```diff
@@ -35,15 +35,31 @@ def setup_arg_parser():
         "--temp", type=float, default=DEFAULT_TEMP, help="Sampling temperature"
     )
     parser.add_argument("--seed", type=int, default=DEFAULT_SEED, help="PRNG seed")
+    parser.add_argument(
+        "--apply_chat_template",
+        "-a",
+        action="store_true",
+        help="apply the tokenizer's chat template on the prompt",
+    )
     return parser
 
 
 def main(args):
     mx.random.seed(args.seed)
     model, tokenizer = load(args.model)
+    if args.apply_chat_template:
+        messages = [{"role": "user", "content": args.prompt}]
+        # test if the tokenizer has a chat template
+        if not hasattr(tokenizer, "apply_chat_template"):
+            raise ValueError(f"{args.model} does not have a chat template")
+        prompt = tokenizer.apply_chat_template(
```

Comment:
Looks like a good improvement. However, could you explain why you would like to disable `add_special_tokens`? I quickly looked at the documentation and it seems that there are more usages such as `tokenizer.apply_chat_template(messages, tokenize=False, add_generation_prompt=True)`.

Reply:
oh I think you're right. My brain has swapped them I suppose ! Good catch. Let me fix that.

Reply:
(for reference, here is the code in the transformers lib https://github.com/huggingface/transformers/blob/3f69f415adcbdaedec154ba8eac220ef3276975d/src/transformers/tokenization_utils_base.py#L1678 )